p2p/channel_test.go

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright (C) 2020 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package p2p
+
+import (
+	"errors"
+	"net"
+	"testing"
+	"time"
+
+	"github.com/mysteriumnetwork/node/core/port"
+	"github.com/mysteriumnetwork/node/p2p/testdata"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestChannelFullCommunicationFlow(t *testing.T) {
+	var err error
+	var consumer, provider *Channel
+	ports := acquirePorts(t, 2)
+	providerPort := ports[0]
+	consumerPort := ports[1]
+	privateKey := GeneratePrivateKey()
+	assert.NoError(t, err)
+
+	t.Run("Test consumer channel creation", func(t *testing.T) {
+		consumer, err = NewChannel(providerPort, privateKey)
+		assert.NoError(t, err)
+		go func() {
```

Comment:
This t.Run already runs in separate goroutine  and since no other code besides ListenAndServe(), theres no point to run inside another goroutine.

Reply:
Yes, but it it still waits for t.Run to finish before running another subtest so I can't block it.